src/main/java/org/alliancegenome/curation_api/auth/AuthenticationFilter.java

```diff
@@ -0,0 +1,166 @@
+        //requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).header(HttpHeaders.WWW_AUTHENTICATE, AUTHENTICATION_SCHEME + " realm=\"" + REALM + "\"").build());
+    }
+
+    private void validateToken(String token) throws Exception {
+        // Check if the token was issued by the server and if it's not expired
+        // Throw an Exception if the token is invalid
+        //log.debug("API Access Token: " + ConfigHelper.getApiAccessToken());
+        //log.info("Validating Token: " + token);
+
+        if(okta_auth.get()) {
+
+            OktaTokenInterface oti = RestProxyFactory.createProxy(OktaTokenInterface.class, okta_url.get());
+
+            String basic = "Basic " + Base64.getEncoder().encodeToString((client_id.get() + ":" + client_secret.get()).getBytes());
+
+            OktaUserInfo info = oti.getUserInfo(basic, "access_token", token);
+
+            Client client = Clients.builder()
+                    .setOrgUrl(okta_url.get())
+                    .setClientId(client_id.get())
+                    .setClientCredentials(new TokenClientCredentials(api_token.get()))
+                    .build();
+
+            User user = client.getUser(info.getUid());
+
+            if(user != null) {
+                SearchResponse<Person> res = personDAO.findPersonByEmail(user.getProfile().getEmail());
+                if(res == null) {
+                    Person person = new Person();
+                    person.setApiToken(UUID.randomUUID().toString());
+                    person.setEmail(user.getProfile().getEmail());
+                    person.setFirstName(user.getProfile().getFirstName());
+                    person.setLastName(user.getProfile().getLastName());
+                    personDAO.persist(person);
+                    userAuthenticatedEvent.fire(person);
+                } else {
+                    userAuthenticatedEvent.fire(res.getResults().get(0));
+                }
+
+            } else {
+                SearchResponse<Person> res = personDAO.findByField("apiToken", api_token.get());
+                if(res != null && res.getResults().size() == 1) {
+                    userAuthenticatedEvent.fire(res.getResults().get(0));
+                } else {
+                    log.warn("Authentication Unsuccessful: " + token);
+                    throw new Exception("Authentication Unsuccessful: " + token + " failed authentication");
+                }
+            }
+        } else {
+            log.warn("OKTA Authentication Disabled using Test Dev User");
+            SearchResponse<Person> res = personDAO.findPersonByEmail("test@alliancegenome.org");
+            if(res == null) {
+                Person person = new Person();
+                person.setApiToken(UUID.randomUUID().toString());
+                person.setEmail("test@alliancegenome.org");
+                person.setFirstName("Local");
+                person.setLastName("Dev User");
+                personDAO.persist(person);
```

Comment:
Does this person needs to persist?

Reply:
Yes... that way it can be used multiple times through local development or integration testing.